code/algorithms.pyx

```diff
@@ -0,0 +1,380 @@
+            if i%3==2 and j==0:
+                u,v = (i,j), (i-1,N2-1)
+                if g.has_edge(u,v):
+                    g.remove_edge(u,v)
+    return g
+
+
+def create_connected_random_graph(N,k):
+    '''Create a connected random graph with N nodes and k edges per node on average.'''
+    #Algorithm: 
+    #First, add nodes and connect them to already connected nodes to ensure that the graph is connected.
+    #Second, add edges between randomly chosen pairs of nodes until the desired number of edges is present.
+    g = nx.Graph()
+    g.add_node(0)
+    nodes = [0]
+    for i in range(1,N):
+        g.add_node(i)
+        g.add_edge(i,random.choice(nodes))
+        nodes.append(i)
+    n_edges = np.array([len(g.edges(n)) for n in g.nodes()])
+    while np.mean(n_edges) <= k:
+        n1,n2 = random.randint(0,N-1), random.randint(0,N-1)
+        if n1 != n2:
+            g.add_edge(n1,n2)
+            n_edges[n1] = n_edges[n1] + 1
+            n_edges[n2] = n_edges[n2] + 1
+    return g
+
+
+def convert_networkx_graph_to_adjacency_list(G):
+    '''Convert a networkx graph G to an adjacency list stored in a numpy array of shape (N,N+1) 
+    in which row i stores the indices of the neighbors of node i followed by -1 values until the end of the row.
+    '''
+    adj_list = - np.ones((G.number_of_nodes(),G.number_of_nodes()+1),dtype=np.int)
+    index_dict = dict([(n,i) for i,n in enumerate(G.nodes_iter())])
+    for node in G.nodes_iter():
+        edges = G.edges(node)
+        adj_list[index_dict[node],:len(edges)] = [index_dict[e[1]] for e in edges]  
+    return adj_list
+
+
+
+
+
+# Function to measure fixation probabilities
+
+def measure_fixation_probabilities(graph_generator, graph_generator_arguments, w, c, n_graphs, n_runs, benefit_to_cost_ratios, dview = None, iteration_type = 1):
+    ''' For each value in benefit_to_cost_ratios, do n_graphs*n_runs simulations to measure the fixation probabilities.
+    
+    Parameters:
+    graph_generator: Function which returns a graph as an adjacency list. This adjacency list should be a numpy array of shape (N,N+1) in which row i stores the indices of the neighbors of node i followed by -1 values until the end of the row.
+    graph_generator_arguments: List of arguments passed to the graph_generator function.
+    w:  Selection strength.
+    c:  Costs.
+    n_graphs: The number of graphs generated with graph_generator.
+    n_runs: The number of simulations on each graph.
+    benefit_to_cost_ratios: List of b/c ratios.
+    dview (optional): ipyparallel.client.view.DirectView object for parallelization. The default value is None, meaning no parallelization.
+    iteration_type (optional): Either 1 (death-birth process) or 0 (imitation process). Default is 1. 
+    
+    Returns:
+    Fixation probabilities as a numpy-array of the same length as benefit_to_cost_ratios.
+    '''
+    fixation_probabilities = []
+    seeds = [rand() for i in xrange(n_graphs)]
```

Comment:
here you create a set of seeds that is then used for graphs of each bc ratio, which means that bc ratio measurements are actually not independent. I do not think it will change a lot, but if the simulation have used random number for the same things in same order it might change something.
You can just move this line inside of the for loop.